modules/rte_mis_allocation/src/Controller/StudentAdmissionReportController.php

```diff
@@ -0,0 +1,721 @@
+          $locationId = $currentUser->get('field_location_details')->getString() ?? NULL;
+          if (!$id) {
+            $url = Url::fromRoute('rte_mis_allocation.controller.student_admission_report', ['id' => $locationId])->toString();
+
+            // Return a redirect response.
+            return new RedirectResponse($url);
+          }
+        }
+      }
+      // Create a table with data.
+      $build = [
+        '#type' => 'table',
+        '#header' => $this->getHeaders($id),
+        '#rows' => $this->getData($id),
+        '#attributes' => ['class' => ['student-reports']],
+        '#cache' => [
+          'contexts' => ['user'],
+          'tags' => [
+            'user_list',
+            'taxonomy_term_list',
+            'mini_node_list',
+          ],
+        ],
+      ];
+
+      return $build;
+
+    }
+    else {
+      throw new NotFoundHttpException();
+    }
+  }
+
+  /**
+   * Function to check if location exists.
+   */
+  protected function checkLocation($id) {
+    $term = $this->entityTypeManager->getStorage('taxonomy_term')->loadByProperties([
+      'vid' => 'location',
+      'tid' => $id,
+      'status' => 1,
+    ]);
+    return (bool) $term;
+  }
+
+  /**
+   * Function to get the headers.
+   */
+  protected function getHeaders($id = NULL) {
+    // Return header based on the user role.
+    $header = [];
+    if ($id) {
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('field_location_details', $id)
+        ->accessCheck(FALSE);
+      $users = $query->execute();
+      $currentUserRole = $users ? $this->entityTypeManager()->getStorage('user')->load(reset($users))->getRoles(TRUE) : [];
+    }
+    else {
+      $currentUserRole = $this->currentUser->getRoles(TRUE);
+    }
+
+    if (array_intersect(['app_admin', 'state_admin'], $currentUserRole)) {
+      $header = ['No.', 'District Name', 'Total Block', 'Total Schools', 'Total RTE seats',
+        'Total Applications', 'Total Applied', 'Total Duplicate', 'Total Incomplete',
+        'Total Rejected', 'Total Approved', 'Total Allotted', 'Total Unallotted', 'Total Admitted',
+        'Total Not Admitted', 'Total Dropped Out',
+      ];
+    }
+    elseif (in_array('district_admin', $currentUserRole)) {
+      $header = ['No.', 'Total Block', 'Total Schools', 'Total RTE seats',
+        'Total Applications', 'Total Applied', 'Total Duplicate', 'Total Incomplete',
+        'Total Rejected', 'Total Approved', 'Total Allotted', 'Total Unallotted', 'Total Admitted',
+        'Total Not Admitted', 'Total Dropped Out',
+      ];
+    }
+    elseif (in_array('block_admin', $currentUserRole)) {
+      $header = ['No.', 'Total Schools', 'Total RTE seats',
+        'Total Applications', 'Total Applied', 'Total Duplicate', 'Total Incomplete',
+        'Total Rejected', 'Total Approved', 'Total Allotted', 'Total Unallotted', 'Total Admitted',
+        'Total Not Admitted', 'Total Dropped Out',
+      ];
+    }
```

Comment:
Please add the string in translatable method.

Reply:
ok